src/js/components/bulkDownload/awards/filters/AwardTypeFilter.jsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * AwardTypeFilter.jsx
+ * Created by Lizzie Salita 11/2/17
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { CheckCircle, ExclamationTriangle } from 'components/sharedComponents/icons/Icons';
+import { indexOf, remove, concat } from 'lodash';
+
+import DownloadCheckbox from '../../DownloadCheckbox';
+
+const propTypes = {
+    awardTypes: PropTypes.array,
+    currentAwardTypes: PropTypes.array,
+    updateFilter: PropTypes.func
+};
+
+export default class AwardTypeFilter extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.onChange = this.onChange.bind(this);
+    }
+
+    onChange(name, value) {
+        if (value) {
+            const updatedAwardTypes = concat(this.props.currentAwardTypes, [name]);
+            this.props.updateFilter('award_types', updatedAwardTypes);
+        }
+        else {
+            const updatedAwardTypes = remove(this.props.currentAwardTypes, (item) => item !== name);
+            this.props.updateFilter('award_types', updatedAwardTypes);
+        }
+    }
+
+    render() {
+        const awardTypesArray = this.props.currentAwardTypes;
+
+        const currentAwardTypes = {
```

Comment:
Why are we handling these types as arrays if we are performing this many search operations with the array? Why not use an object or other keyed data structure?

Reply:
Same comment applies to the above `onChange` function